compiler/noirc_frontend/src/hir/comptime/interpreter/foreign.rs

```diff
@@ -53,21 +132,190 @@ pub(super) fn apply_range_constraint(
     }
 }
 
-// poseidon2_permutation<let N: u32>(_input: [Field; N], _state_length: u32) -> [Field; N]
+/// `fn from_le_bytes(bytes: [u8], modulus: [u8]) -> BigInt`
+///
+/// Returns the ID of the new bigint allocated by the solver.
+fn bigint_from_le_bytes(
+    interner: &mut NodeInterner,
+    solver: &mut BigintSolverWithId,
+    arguments: Vec<(Value, Location)>,
+    location: Location,
+) -> IResult<Value> {
+    let (bytes, modulus) = check_two_arguments(arguments, location)?;
+
+    let (bytes, _) = get_slice_map(interner, bytes, get_u8)?;
+    let (modulus, _) = get_slice_map(interner, modulus, get_u8)?;
+
+    let id = solver
+        .bigint_from_bytes(&bytes, &modulus)
+        .map_err(|e| InterpreterError::BlackBoxError(e, location))?;
+
+    Ok(Value::U32(id))
```

Comment:
How does this work if we need to return a BigInt value rather than an ID to one?
Don't we need to return a struct value: `BigInt { pointer: ?, modulus: ? }`?

As-is it looks like users would be able to do `let new_bigint = my_bigint + 2` and offset the id pointer

Reply:
A fun (unfun?) trick is that you can probably trick the type system using a macro:
```rs
comptime {
    let a = bigint::Secpk1Fq::from_le_bytes(&[9]);
    let b = bigint::BigInt::from_le_bytes(a.to_le_bytes(), bigint::bn254_fq);

    let b: u32 = std::meta::unquote!(quote { $b });
    println(b + 2);
}
```
That should probably work but it's not a big deal in that case since it's somewhat expected to be a type system hole anyway.

I think a case that would fail would be if a user or library author tried to access a struct field on the U32 id value.

Reply:
Thanks for pointing out the holes! 😃 
Yes, like I mentioned in the PR a valid use case would IMO be to add a `clone()` method to `BigInt` that seems reasonable to stay on Noir side and copy the two fields. Best to go with a Struct.

Reply:
Changed to construct `BigInt` based on the `return_type`.

Reply:
@michaeljklein fyi on the above as you mentioned looking for hacky ways to convert between types for the enum experiment.